src/tools/cephfs/DataScan.cc

```diff
@@ -1358,7 +1358,7 @@ int DataScan::scan_links()
       derr << "Unexpected error reading dentry "
 	   << p.second.dirfrag() << "/" << p.second.name
 	   << ": " << cpp_strerror(r) << dendl;
-      return r;
+      continue;
```

Comment:
is this dentry(which could not be read) queued for re-reading?

Reply:
No.

Reply:
okay, but is it fine to not do it? you want the user to re-run the data scan tool?

Reply:
What do we get by queuing for retry? If the dentry is missing, who do we expect to create it?

Reply:
the `derr` line says "Unexpected error reading dentry", might be worth re-trying once i guess if something intermittently failed while scanning? or if we are sure for the fact that dentry is missing, derr can be changed to something "dentry foo missing"

Reply:
Logging to stderr has value since it the entry read failure is not expected (something removed it, maybe a bug somewhere), but the tool can still continue with other inodes/dentries.

Honestly, I'm not sure retrying would be of much value - retry once and if that fails - we would want to still move forward, so why not do that anyway.

Reply:
makes sense, it was just the error line "Unexpected error reading dentry" that made me think of re-trying